isisd/isis_sr.c

```diff
@@ -655,67 +950,94 @@ static void isis_sr_prefix_uninstall(struct sr_prefix *srp)
 	}
 }
 
-/* Reinstall local or remote Prefix-SID. */
-static void isis_sr_prefix_reinstall(struct sr_prefix *srp,
-				     bool replace_semantics)
+/**
+ * Reinstall local or remote Prefix-SID.
+ *
+ * @param srp	Segment Routing Prefix
+ */
+static inline void sr_prefix_reinstall(struct sr_prefix *srp)
 {
 	/*
-	 * Route replace semantics can be used only when we know for sure that
-	 * the Prefix-SID input label hasn't changed. Otherwise we need to
-	 * uninstall the Prefix-SID first using the old input label before
-	 * reinstalling it.
+	 * The Prefix-SID input label has changed. We need to un-install the
+	 * Prefix-SID first using the old input label before reinstalling it.
 	 */
-	if (!replace_semantics)
-		isis_sr_prefix_uninstall(srp);
+	sr_prefix_uninstall(srp);
```

Comment:
Because, I think it is better to add an explicit comment when using the Replace Semantics (Make before break) instead of simply calling the reinstall function with true or false parameters. So, the modification consist to call reinstall function that uninstall and install Label by default and explicitly call install function when Replace Semantics could be used. In addition, this could be use only in one place and I add an explicit comment when it install function is used instead of reinstall function.

Reply:
> So, the modification consist to call reinstall function that uninstall and install Label by default

This is exactly what we should avoid doing for the reason I explained before. We can only have make-before-break behavior if the Prefix-SIDs are reinstalled by sending a single `ZEBRA_MPLS_LABELS_REPLACE` message to zebra.

Reply:
OK. Understand. I revert the patch, but change variable name from 'replace_semantics' to 'make_before_break' and update comments. I think that 'Make Before Break' terminology is more adapted